### PR TITLE
Added better cross-browser support for the visibilitychange event in GWT

### PR DESCRIPTION
**Pull request by barkholt**
Support the visibilitychange event on a wider range of desktop and mobile browsers

### CODE REVIEW

File: backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtApplication.java [hunk 1]
```diff
@@ -538,11 +538,28 @@ native static public void consoleLog(String message) /*-{
...
+		} else if ("webkitHidden" in $doc) {
+			eventName = "webkitvisibilitychange"
+		} else if ("mozHidden" in $doc) {
+			eventName = "mozvisibilitychange"
+		} else if ("msHidden" in $doc) {
+			eventName = "msvisibilitychange"
+		}
+
+		if (eventName !== null) {
+			$doc
```
Comment (intrigus) [inline backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtApplication.java]: Could you maybe fix this formatting?

Comment (barkholt) [inline backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtApplication.java]: The formatting has been done automatically by the Eclipse formatter standard libgdx has supplied. Which part do you consider in need of fixing?

File: backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtApplication.java [hunk 2]
```diff
@@ -538,11 +538,28 @@ native static public void consoleLog(String message) /*-{
...
+					.addEventListener(
+							eventName,
+							function(e) {
+								self.@com.badlogic.gdx.backends.gwt.GwtApplication::onVisibilityChange(Z)($doc['hidden'] !== true);
+							});
+		}
```
Comment (Tom-Ski) [inline backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtApplication.java]: The lines above this comment. The formatter probably doesn't play nice with jsni so best to do it manually according to the usual libgdx formatting for java.

**Comment by barkholt** [comment]
Closing this, and replaced it with #4275, due to me messing up the squashing of commits

**Pull request closed without merging**